### ISSUE TITLE
Sideloaded child records in adapter response not persisted in "reload" setting

### ISSUE THREAD

**Issue by samcic** [emberjs/data#3580]
JSbin:

http://emberjs.jsbin.com/robafonika/1/edit?html,css,js,console,output

I'm not 100% sure if this is "by design" or not, but it seems somewhat counter-intuitive. 

Scenario:
* create a record in a route's model hook, save it and set that record as my route model, 
* in the template, edit an attribute of a child of the model,
* then "refresh" the entire route model by simply creating another record and saving it (getting back a record and child record with the same id as the first save operation), 

So, in short: the "server" response doesn't seem to be updating the child record (i.e. edit the 'test2' value, press enter and see it NOT get overwritten). It does update the parent record (i.e. edit the 'test1' value, press enter and see it get overwritten).

**Comment by wecc** [comment]
Hi @samcic, thanks for reporting!

From what I can tell this is what's happening in your JSBin:

1. You create a new parent, save it and return data + child relationship. There is now one parent and one child in the store.
2. When you edit the parent you create a new parent, save it and return data + child relationship. There is now two parents and one child in the store. Both parents' `myattr`s has the canonical value `test1` but the original parent also has a modified value (of whatever you modified it to). But your model is not the original parent anymore, it's the new one.
3. When you edit the child you do the same thing as above. The difference is that the child model is still referencing the original child, which means that even though the canonical value of the child's `anotherattr` is `test2` you haven't saved the child so the child still has a modified attribute and is in a dirty state.

In general I think this behavior is intended but I do however think we have a bug here since you shouldn't really be able to assign the same ID to two different records. I think this might have something to do with how Model and InternalModel handles IDs. @igorT thoughts?

**Comment by samcic** [comment]
Hi @wecc, cheers for the info! I didn't consider the dirty aspect of the child, so that makes much more sense now. As you said, it does however seem strange that there can be two (or more?) records existing, both with the same type and id.

For the record (yep, best pun of my day so far), I think you guys are doing a sensational job with this library overall. The Ember buffet (ember/cli/data) is truly delicious these days.

**Comment by wecc** [comment]
@samcic thank you for your kind words! I'm closing this as there has been work on preventing multiple records with the same ID in a couple of other issues since this was reported.
